From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 17 Dec 2018 18:24:36 +0000
Subject: x86/mm: Really use WRITE_ONCE() when setting PTEs
Forwarded: not-needed

Commit 1f07011afbf6 "x86/mm: Use WRITE_ONCE() when setting PTEs"
actually uses ACCESS_ONCE() because WRITE_ONCE() isn't defined on
the 3.16-stable branch.

We do have a definition of WRITE_ONCE() and also stricter rules for
using ACCESS_ONCE().  So use WRITE_ONCE() here.

---
--- a/arch/x86/include/asm/pgtable_64.h
+++ b/arch/x86/include/asm/pgtable_64.h
@@ -46,7 +46,7 @@ void set_pte_vaddr_pud(pud_t *pud_page,
 
 static inline void native_set_pte(pte_t *ptep, pte_t pte)
 {
-	ACCESS_ONCE(*ptep) = pte;
+	WRITE_ONCE(*ptep, pte);
 }
 
 static inline void native_pte_clear(struct mm_struct *mm, unsigned long addr,
@@ -62,7 +62,7 @@ static inline void native_set_pte_atomic
 
 static inline void native_set_pmd(pmd_t *pmdp, pmd_t pmd)
 {
-	ACCESS_ONCE(*pmdp) = pmd;
+	WRITE_ONCE(*pmdp, pmd);
 }
 
 static inline void native_pmd_clear(pmd_t *pmd)
@@ -98,7 +98,7 @@ static inline pmd_t native_pmdp_get_and_
 
 static inline void native_set_pud(pud_t *pudp, pud_t pud)
 {
-	ACCESS_ONCE(*pudp) = pud;
+	WRITE_ONCE(*pudp, pud);
 }
 
 static inline void native_pud_clear(pud_t *pud)
@@ -131,7 +131,7 @@ static inline pgd_t *native_get_shadow_p
 
 static inline void native_set_pgd(pgd_t *pgdp, pgd_t pgd)
 {
-	ACCESS_ONCE(*pgdp) = kaiser_set_shadow_pgd(pgdp, pgd);
+	WRITE_ONCE(*pgdp, kaiser_set_shadow_pgd(pgdp, pgd));
 }
 
 static inline void native_pgd_clear(pgd_t *pgd)
